backend/persistent/datastores/livy.js

```diff
@@ -0,0 +1,208 @@
+import fetch from 'node-fetch';
+
+import {dissoc} from 'ramda';
+
+import Logger from '../../logger';
+
+export function connect(connection) {
+    Logger.log('' +
+        'Attempting to authenticate with connection ' +
+        `${JSON.stringify(dissoc('password', connection), null, 2)} ` +
+        '(password omitted)'
+    );
+
+    return newSession(connection);
+}
+
+export function tables(connection) {
+    const code = (connection.database) ?
+        `val r = plotlyContext.sql("show tables in ${connection.database}").collect\n%json r` :
+        `val r = plotlyContext.sql("show tables").collect\n%json r`;
+    return sendRequest(connection, code)
+        .then(json => {
+            if (!Array.isArray(json) || json.length <= 0) {
```

Comment:
why `<`? can this just be `=== 0`?

Reply:
It's just a style thing. I'll change it to `===`.